src/lib/kademlia/node_addrs_and_ports.ml

```diff
@@ -8,7 +8,33 @@ type t =
   ; client_port: int
   ; libp2p_port: int
   ; communication_port: int }
-[@@deriving bin_io]
+[@@deriving bin_io, fields]
+
+module Display = struct
+  module Stable = struct
+    module V1 = struct
+      type t =
+        { external_ip: string
+        ; bind_ip: string
+        ; discovery_port: int
+        ; client_port: int
+        ; libp2p_port: int
+        ; communication_port: int }
+      [@@deriving fields, yojson, bin_io]
+    end
+  end
+end
+
+let to_display (t : t) =
+  Display.Stable.V1.
```

Comment:
This should use `Display` instead of the specific version, so it's always on the latest version if anything changes.